src/overlays/actors/ovl_Elf_Msg3/z_elf_msg3.c

```diff
@@ -29,28 +29,122 @@ const ActorInit Elf_Msg3_InitVars = {
+        if (ABS_ALT(this->actor.home.rot.x) == 0) {
+            this->actor.scale.z = 0.4f;
+            this->actor.scale.x = 0.4f;
+        } else {
+            absolute = ABS_ALT(this->actor.home.rot.x);
+            this->actor.scale.z = absolute * 0.04f;
+            this->actor.scale.x = absolute * 0.04f;
+        }
+        if (this->actor.home.rot.z == 0) {
+            this->actor.scale.y = 0.4f;
+        } else {
+            this->actor.scale.y = this->actor.home.rot.z * 0.04f;
+        }
+        ElfMsg3_SetupAction(this, func_80A2CF7C);
+        this->actor.shape.rot.z = 0;
+        this->actor.shape.rot.y = this->actor.shape.rot.z;
+        this->actor.shape.rot.x = this->actor.shape.rot.z;
+    }
+}
+
+void ElfMsg3_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+s32 func_80A2CF50(ElfMsg3* this) {
+    if (ELFMSG3_GET_8000(&this->actor) != 0) {
+        return (ELFMSG3_GET_FF(&this->actor)) + 0x200;
+    }
+    return -0x200 - (ELFMSG3_GET_FF(&this->actor));
+}
+
+void func_80A2CF7C(ElfMsg3* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    EnElf* elf = (EnElf*)player->tatlActor;
+
+    if (((((player->tatlActor != NULL) &&
+           (fabsf(player->actor.world.pos.x - this->actor.world.pos.x) < (100.0f * this->actor.scale.x))) &&
+          (this->actor.world.pos.y <= player->actor.world.pos.y)) &&
+         ((player->actor.world.pos.y - this->actor.world.pos.y) < (100.0f * this->actor.scale.y))) &&
+        (fabsf(player->actor.world.pos.z - this->actor.world.pos.z) < (100.0f * this->actor.scale.z))) {
+        player->tatlTextId = func_80A2CF50(this);
+        ActorCutscene_SetIntentToPlay(0x7C);
+        elf->elfMsg = &this->actor;
+        if (this->actor.cutscene == -1) {
+            this->actor.cutscene = 0x7C;
+        }
+        if ((player->tatlTextId < 0) && (this->actor.home.rot.x < 0)) {
+            if (ActorCutscene_GetCurrentIndex() == 0x7D) {
+                ActorCutscene_Stop(0x7D);
+                ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+            } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene) != 0) {
+                ActorCutscene_Start(this->actor.cutscene, &this->actor);
+                func_800E0348(globalCtx->cameraPtrs[0]);
+            } else {
+                ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+            }
+        }
+    }
+}
+
+void ElfMsg3_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ElfMsg3* this = THIS;
+
+    if (func_80A2CD1C(this, globalCtx) == 0) {
+        if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state) != 0) {
+            if (ELFMSG3_GET_SWITCH(thisx) != 0x7F) {
+                Flags_SetSwitch(globalCtx, ELFMSG3_GET_SWITCH(thisx));
+            }
+            Actor_MarkForDeath(&this->actor);
+        } else if ((this->actor.home.rot.y >= 0) || (this->actor.home.rot.y < -0x80) ||
+                   (Flags_GetSwitch(globalCtx, -1 - this->actor.home.rot.y) != 0)) {
```

Comment:
```suggestion
                   Flags_GetSwitch(globalCtx, -1 - this->actor.home.rot.y)) {
```
Weird access, but this is a weird actor.